table/block_based_table_reader.cc

```diff
@@ -2162,6 +2109,64 @@ Status BlockBasedTable::MaybeReadBlockAndLoadToCache(
   return s;
 }
 
+Status BlockBasedTable::RetrieveBlock(
+    FilePrefetchBuffer* prefetch_buffer, const Rep* rep, const ReadOptions& ro,
+    const BlockHandle& handle, const UncompressionDict& uncompression_dict,
+    CachableEntry<Block>* block_entry, bool is_index, GetContext* get_context) {
+
+  assert(rep);
+  assert(block_entry);
+  assert(block_entry->IsEmpty());
+
+  Status s;
+  if (!is_index || rep->table_options.cache_index_and_filter_blocks) {
```

Comment:
Should we check for non-null compressed and uncompressed block cache, rather than ```rep->table_options.cache_index_and_filter_blocks```?

Reply:
Well, it is possible that the customer has a block cache configured but they don't want to use it for the index/filter block. So I think this is the right check. As for the null checks, MaybeReadBlockAndLoadToCache performs those internally.